### ISSUE TITLE
Why PyCrypto for RSA?

### ISSUE THREAD

**Issue by TkTech** [idan/oauthlib#226]
About two years ago it appears pycrypto was swapped out for python-rsa. Then at some point about a year ago pycrypto was re-introduced (this time as a soft dependency).

My questions:
- Why was this decision made?
- If python-rsa was no longer viable, why not the alternatives like tlslite?
- Why is there no easy extension mechanism for swapping out implementations?

I ask because [jira-python][] already includes [tlslite][] which is perfectly capable of doing the signing, but is pure-python and can optionally use pycrypto for acceleration. Howerver, there is no easy (or maybe there is but undocumented?) way of swapping it out without forking the code. I had to resort to monkey patching to do so:

```python
def _pure_python_rsa_sha1(base_string, rsa_private_key):
    """
    An alternative, pure-python RSA-SHA1 signing method, using
    the tlslite library.
    """
    import base64
    from tlslite.utils import keyfactory

    private_key = keyfactory.parsePrivateKey(rsa_private_key)
    signature = private_key.hashAndSign(base_string.encode('ascii'))

    # This *must* be unicode, since oauthlib refuses to work with
    # anything else.
    return unicode(base64.b64encode(signature))

# Monkey patch the built-in RSA-SHA1 signing method, since the
# package offers no proper extension mechanism.
from oauthlib.oauth1.rfc5849 import signature
signature.sign_rsa_sha1 = _pure_python_rsa_sha1
```

[jira-python]: http://jira-python.readthedocs.org/en/latest/
[tlslite]: https://pypi.python.org/pypi/tlslite

**Comment by Zopieux** [comment]
@ib-lundgren Just so you know, [pyca/cryptography#1331](https://github.com/pyca/cryptography/pull/1331) has been merged.

**Comment by thedrow** [comment]
Woot!

**Comment by tilgovi** [comment]
Can be closed now that #322 merged.

**Comment by thedrow** [comment]
@ib-lundgren ping. You can close this issue now.
